File: editor/src/de/fatox/meta/ide/SceneManager.java
package de.fatox.meta.ide;

import com.badlogic.gdx.files.FileHandle;
import de.fatox.meta.api.model.MetaSceneData;
import de.fatox.meta.shader.MetaSceneHandle;

/**
 * Created by Frotty on 15.06.2016.
 */
public interface SceneManager {

    MetaSceneHandle createNew(String name);

    void loadScene(FileHandle projectFile);

    void saveScene(MetaSceneData sceneData);
}

File: editor/src/main/kotlin/de/fatox/meta/ide/SceneManager.kt
package de.fatox.meta.ide

import com.badlogic.gdx.files.FileHandle
import de.fatox.meta.shader.MetaSceneHandle

interface SceneManager {
	fun createNew(name: String): MetaSceneHandle
	fun loadScene(projectFile: FileHandle)
	fun saveScene(sceneHandle: MetaSceneHandle)
}